### PR TITLE
Update DeepStream and GMSL

### PR DESCRIPTION
**Pull request by DanielHo-BS**
1. Update DeepStream
  1. Add deepstream-demo.png
  2. Fix the English grammar

2. Update GMSL
  1. nvidia/gmsl_camera/images
  2. nvidia/gmsl_camera/index.rst
  3. nvidia/gmsl_camera/system_diagram.rst
  4. nvidia/gmsl_camera/camera_driver.rst
  5. nvidia/gmsl_camera/test_result.rst
  6. nvidia/gmsl_camera/frame_sync.rst

### CODE REVIEW

File: nvidia/gmsl_camera/camera_driver.rst [hunk 1]
```diff
@@ -3,4 +3,76 @@
 Camera Driver Usage
 ###################
 
-TODO
+You need to install dedicated Camera's driver (e.g. ``leopard-ar0233-gmsl``).
+
+The driver is provided by **ADLINK** and could not be used on other vendor's platform since it is compiled into binary.
+
+And it can only download form **ADLINK ROScube's PPA server**.
```
Comment (evshary) [inline nvidia/gmsl_camera/camera_driver.rst]: PPA server -> APT repository

File: nvidia/gmsl_camera/camera_driver.rst [hunk 2]
```diff
@@ -3,4 +3,76 @@
 Camera Driver Usage
```
Comment (evshary) [inline nvidia/gmsl_camera/camera_driver.rst]: "Camera Driver Installation" might be better.

File: nvidia/gmsl_camera/index.rst [hunk 3]
```diff
@@ -3,14 +3,29 @@
...
 **Table of Contents:**
 
 .. toctree::
    :maxdepth: 1
 
+   system_diagram.rst
    camera_driver.rst
+   test_result.rst
+   frame_sync.rst
+.. TODO: Will be added in the future
```
Comment (evshary) [inline nvidia/gmsl_camera/index.rst]: The following TODO can be removed. I think your documents are enough now.

File: nvidia/gmsl_camera/test_result.rst [hunk 4]
```diff
@@ -3,4 +3,54 @@
 Test Result
 ###########
 
-TODO
+Cameras without ISP can use **Argus API** to preview the camera's video streaming. 
+
+Or you can use **GStreamer NVArgusCameraSrc plugin** to preview the video streaming.
+
+1. Video Frame Preview by GStreamer.
```
Comment (evshary) [inline nvidia/gmsl_camera/test_result.rst]: There is only 1 step, so 1. might not be necessary.

File: nvidia/gmsl_camera/frame_sync.rst [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+| But every camera has different clock source, so they will not able to output the frame at the same time.
+
+| Like the following plot.
+| Camera's frame 1 (f1) and camera's frame 2 (f2) is not at the same time, therefore, under your system they are not under the same timestamp.
+
+.. image:: images/free-run-30fps.png
+  :width: 80%
+  :align: center
+
+| Frame Sync mode is a mechanism to make sure all of your camera is in passive mode, and every frame can by triggered by external signal.
```
Comment (evshary) [inline nvidia/gmsl_camera/frame_sync.rst]: all of your camera"s" "are" in passive mode,

File: nvidia/gmsl_camera/frame_sync.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+| Like the following plot.
+| Camera's frame 1 (f1) and camera's frame 2 (f2) is not at the same time, therefore, under your system they are not under the same timestamp.
+
+.. image:: images/free-run-30fps.png
+  :width: 80%
+  :align: center
+
+| Frame Sync mode is a mechanism to make sure all of your camera is in passive mode, and every frame can by triggered by external signal. 
+| Therefore, host side has the full permission to control when the camera to collect the frame and the received timestamp.
```
Comment (evshary) [inline nvidia/gmsl_camera/frame_sync.rst]: when the camera collects the frame and the received timestamp.

File: nvidia/gmsl_camera/frame_sync.rst [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    cat /sys/module/leopard-ar0233/parameters/trigger_mode
+
+.. image:: images/ar0233-trigger-mode.png
+  :width: 80%
+  :align: center
+
+
+.. note::
+
+    If value is **0**, which means all AR0233 camera is in free run-mode, if value is **1**, all ar0233 cameras are in frame sync mode.
```
Comment (evshary) [inline nvidia/gmsl_camera/frame_sync.rst]: AR0233 camera"s" are
ar0233 => AR0233

File: nvidia/gmsl_camera/frame_sync.rst [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+2. Trigger Frames by external I/O
+---------------------------------
+
+Now we can use external I/O library to control FSYNC frequence. ROScube has provides a I/O library called **Neuron Library**.
```
Comment (evshary) [inline nvidia/gmsl_camera/frame_sync.rst]: ROScube has provides => ADLINK provides

**Review by evshary — CHANGES_REQUESTED** [review]

File: nvidia/gmsl_camera/camera_driver.rst (hunk 1 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/camera_driver.rst]: Done

File: nvidia/gmsl_camera/camera_driver.rst (hunk 2 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/camera_driver.rst]: Yes

File: nvidia/gmsl_camera/index.rst (hunk 3 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/index.rst]: OK

File: nvidia/gmsl_camera/test_result.rst (hunk 4 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/test_result.rst]: OK

File: nvidia/gmsl_camera/frame_sync.rst (hunk 5 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/frame_sync.rst]: Done

File: nvidia/gmsl_camera/frame_sync.rst (hunk 6 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/frame_sync.rst]: Done

File: nvidia/gmsl_camera/frame_sync.rst (hunk 7 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/frame_sync.rst]: Done

File: nvidia/gmsl_camera/frame_sync.rst (hunk 8 above)
Comment (DanielHo-BS) [reply nvidia/gmsl_camera/frame_sync.rst]: Done

**Review by DanielHo-BS — COMMENTED** [review]
Finished these request changes.

**Comment by evshary** [comment]
LGTM

**Pull request merged**